### PR TITLE
[dv/doc] security verification in cip_lib

### PR DESCRIPTION
**Pull request by weicaiyang**
Add a section - security verification in cip_lib in cip_lib doc to
describes the steps to enabling tests for countermeasures

Will add following items soon
1. Security Verification for MUBI type CSRs/ports
2. methodology docs of security verification for common countermeasure
primitives

Signed-off-by: Weicai Yang <weicai@google.com>

### CODE REVIEW

File: hw/dv/sv/cip_lib/doc/index.md [hunk 1]
```diff
@@ -393,3 +393,150 @@ together the static side `tb` which instantiates the `dut`, and the dynamic
...
+If the block uses common security countermeasure primitives (prim_count, prim_sparse_fsm_flop, prim_double_lfsr), users can enable this sequence to fully verify them via following steps.
+The details of the sequences can be found in testplans - [sec_cm_count_testplan](https://github.com/lowRISC/opentitan/blob/master/hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson), [sec_cm_fsm_testplan](https://github.com/lowRISC/opentitan/blob/master/hw/dv/tools/dvsim/testplans/sec_cm_fsm_testplan.hjson) and [sec_cm_double_lfsr_testplan](https://github.com/lowRISC/opentitan/blob/master/hw/dv/tools/dvsim/testplans/sec_cm_double_lfsr_testplan.hjson).
+
+1. Import the applicable sec_cm testplans.
+If more checks or sequences are needed, add another testpoint in the block testplan.
+For example, when the fault is detected by countermeasure, the following operation won’t be executed.
```
Comment (matutem) [inline hw/dv/sv/cip_lib/doc/index.md:449]:
```suggestion
For example, when the fault is detected by countermeasure, some subsequent operation won’t be executed.
```

File: hw/dv/sv/cip_lib/doc/index.md [hunk 2]
```diff
@@ -393,3 +393,150 @@ together the static side `tb` which instantiates the `dut`, and the dynamic
...
+4. Set alert name for countermeasure if the alert name is different from the default name - “fatal_fault”.
+```systemverilog
+class ip_env_cfg extends cip_base_env_cfg #(.RAL_T(ip_reg_block));
+  virtual function void initialize(bit [31:0] csr_base_addr = '1);
+    super.initialize(csr_base_addr);
+    sec_cm_alert_name = "fatal_check_error";
+```
+
+5. Override the `check_sec_cm_fi_resp` task in ip_common_vseq to add additional sequences and checks after fault injection.
+This is the example from keymgr, in which CSR `fault_status` will be updated according to the location of the fault and the operation after fault inject will lead design to enter `StInvalid` state.
```
Comment (matutem) [inline hw/dv/sv/cip_lib/doc/index.md:480]:
```suggestion
This is an example from keymgr, in which CSR `fault_status` will be updated according to the location of the fault and the operation after fault inject will lead design to enter `StInvalid` state.
```

**Review by matutem — APPROVED** [review]
Thanks for documenting this! I left some suggestions, but overall LGTM.

File: hw/dv/sv/cip_lib/doc/index.md [hunk 3]
```diff
@@ -393,3 +393,150 @@ together the static side `tb` which instantiates the `dut`, and the dynamic
 side, which is the UVM environment extended from CIP library. The diagram
 lists some common items that need to be instantiated in `tb`
 and set into `uvm_config_db` for the testbench to work.
+
+## Security Verification in cip_lib
+CIP contains reusable security verification components, sequences and function coverage.
+This section describes the details of them and the steps to enabling them.
```
Comment (cindychip) [inline hw/dv/sv/cip_lib/doc/index.md:399]: I am actually not sure which one is grammarly correct - 
```suggestion
This section describes the details of them and the steps to enable them.
```

**Review by cindychip — APPROVED** [review]

**Comment by weicaiyang** [comment]
Thanks @matutem and @cindychip's review. Updated as you suggested.

**Pull request merged**